src/server/lib/middleware/requestContext.ts

```diff
@@ -0,0 +1,36 @@
+import { NextFunction, Request, Response } from 'express';
+import { AsyncLocalStorage } from 'node:async_hooks';
+
+type LoggerContext = {
+	request: Request;
+	response: Response;
+};
+
+export const requestContext = new AsyncLocalStorage<LoggerContext>();
```

Comment:
Whats the maximum storage size of an `AsyncLocalStorage`? Does it keep growing for every request? Or is node smart enough to cleanup/garbage collect once the request is over?

Just worried about this filling up potentially over a period of time, and then the instance crashing and restarting because it was out of memory.

Reply:
The documentation suggests that it is cleared up when the request and hence function chain is over, which is great!

And also this StackOverflow answer: https://stackoverflow.com/a/78392911